### PR TITLE
[Service Fabric Managed Clusters] adding api version 2020-05-01

### PR DESCRIPTION
**Pull request by a-santamaria**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/nodetype.json [hunk 1]
```diff
@@ -0,0 +1,1003 @@
...
+    "DiskType": {
+      "type": "string",
+      "description": "Managed data disk type. IOPS and throughput are given by the disk size, to see more information go to https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types.\n\n  - Standard_LRS - standard HDD LRS\n  - StandardSSD_LRS - standard SSD LRS\n  - Premium_LRS - Premium SSD LRS\n",
+      "enum": [
+        "Standard_LRS",
+        "StandardSSD_LRS",
+        "Premium_LRS"
+      ],
+      "default": "StandardSSD_LRS"
+    },
```
Comment (zcxcasd) [inline specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/nodetype.json:492]: need x-ms-enum

Comment (a-santamaria) [reply specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/nodetype.json:510]: thanks fixed, I also added managedClusterVersions paths that I was missing

Comment (a-santamaria) [reply specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/nodetype.json:510]: @zcxcasd could you review again?

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json [hunk 2]
```diff
@@ -0,0 +1,1368 @@
...
+          }
+        },
+        "tags": [
+          "ManagedClusterVersion"
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/ManagedClusterCodeVersionListResult"
```
Comment (majastrz) [inline specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json:357]: >ManagedClusterCodeVersionListResult [](start = 37, length = 35)

Why is the path to get a specific cluster version returning a list?

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json [hunk 3]
```diff
@@ -0,0 +1,1368 @@
...
```
Comment (majastrz) [inline specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json:403]: >ManagedClusterCodeVersionListResult [](start = 37, length = 35)

Why is the path to get a specific cluster version returning a list? Did you mean a single version here?

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json [hunk 4]
```diff
@@ -0,0 +1,1368 @@
...
+        "issuerThumbprint": {
+          "type": "string",
+          "description": "Issuer thumbprint for the certificate. Only used together with CommonName."
+        }
+      },
+      "description": "Client certificate definition."
+    },
+    "ClusterState": {
+      "type": "string",
+      "description": "The current state of the cluster.\n\n  - WaitingForNodes - Indicates that the cluster resource is created and the resource provider is waiting for Service Fabric VM extension to boot up and report to it.\n  - Deploying - Indicates that the Service Fabric runtime is being installed on the VMs. Cluster resource will be in this state until the cluster boots up and system services are up.\n  - BaselineUpgrade - Indicates that the cluster is upgrading to establishes the cluster version. This upgrade is automatically initiated when the cluster boots up for the first time.\n  - Upgrading - Indicates that the cluster is being upgraded with the user provided configuration.\n  - UpgradeFailed - Indicates that the last upgrade for the cluster has failed.\n  - Ready - Indicates that the cluster is in a stable state.\n",
```
Comment (majastrz) [inline specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json:607]: >description [](start = 7, length = 11)

The x-ms-enum extension allows you to specify a description for each value of the enumeration. See Microsoft.Blueprint swagger for an example.

Comment (majastrz) [reply specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json:607]: Actually never mind. See `ClusterUpgradeMode` for an example.

---
In reply to: [612735158](https://github.com/Azure/azure-rest-api-specs/pull/13860#discussion_r612735158) [](ancestors = 612735158)

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json [hunk 5]
```diff
@@ -0,0 +1,1368 @@
...
+    "api-version": {
+      "name": "api-version",
+      "in": "query",
+      "description": "The version of the Service Fabric resource provider API. This is a required parameter and it's value must be \"2021-05-01\" for this specification.",
+      "required": true,
+      "type": "string",
+      "enum": [
+        "2021-05-01"
+      ],
+      "default": "2021-05-01",
```
Comment (majastrz) [inline specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json:1313]: >default [](start = 7, length = 7)

nit - It doesn't harm anything, but `api-version` is a required parameter, so the default value isn't useful.

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json (hunk 2 above)
Comment (a-santamaria) [reply specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json]: fixed

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json (hunk 3 above)
Comment (a-santamaria) [reply specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json]: yeah this was a mistake, I meant to have a single version, fixed.

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json (hunk 4 above)
Comment (a-santamaria) [reply specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json]: fixed thanks

File: specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json (hunk 5 above)
Comment (a-santamaria) [reply specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json]: ok removed thanks

**Review by majastrz — APPROVED** [review]
signed off from ARM side.

**Comment by a-santamaria** [comment]
@njuCZ can you merge the pr please it has been signed of by arm, we want to generate the sdks and modify the Powershell and CLI clients before the code complete deadline for the next release at the end of the week.

**Review by njuCZ — APPROVED** [review]

**Pull request merged**